tecton_client/_internal/tecton_client.py

```diff
@@ -0,0 +1,102 @@
+                such as default timeout or connection settings. See https://www.python-httpx.org/ for more info.
+        """
+        self.url = url
+        self.default_workspace_name = default_workspace_name
+        self._api_key = api_key
+        self._base_url = urljoin(url, "/api/v1/")
+
+        headers = get_default_headers(api_key)
+        if client is None:
+            self._client = httpx.Client(headers=headers)
+        else:
+            self._client = client
+            # add the headers to the existing client headers
+            self._client.headers.update(headers)
+
+    def get_features(
+        self,
+        *,
+        feature_service_name: Optional[str] = None,
+        feature_service_id: Optional[str] = None,
+        join_key_map: Optional[Dict[str, Union[int, str, type(None)]]] = None,
+        request_context_map: Optional[Dict[str, Any]] = None,
+        metadata_options: Optional[Dict[str, bool]] = None,
+        workspace_name: Optional[str] = None,
+        request_options: Optional[Dict[str, bool]] = None,
+        allow_partial_results: bool = False,
+    ) -> GetFeaturesResponse:
+        validate_request_args(feature_service_id, feature_service_name, workspace_name, self.default_workspace_name)
+        if not workspace_name:
+            workspace_name = self.default_workspace_name
+        request_data = build_get_features_request(
+            feature_service_id=feature_service_id,
+            feature_service_name=feature_service_name,
+            join_key_map=join_key_map,
+            request_context_map=request_context_map,
+            workspace_name=workspace_name,
+            metadata_options=metadata_options,
+            allow_partial_results=allow_partial_results,
+            request_options=request_options,
+        )
+        resp = self._client.post(url=urljoin(self._base_url, "feature-service/get-features"), json=request_data)
+
+        try:
+            resp.raise_for_status()
+        except HTTPStatusError as exc:
+            raise convert_exception(exc) from exc
+
+        return GetFeaturesResponse.from_dict(json.loads(resp.text))
+
+    def get_feature_service_metadata(
+        self,
+        *,
+        feature_service_name: Optional[str] = None,
+        feature_service_id: Optional[str] = None,
+        workspace_name: Optional[str] = None,
+    ) -> GetFeatureServiceMetadataResponse:
+        validate_request_args(feature_service_id, feature_service_name, workspace_name, self.default_workspace_name)
+        if not workspace_name:
+            workspace_name = self.default_workspace_name
+        request_data = build_get_feature_service_metadata_request(
+            feature_service_id=feature_service_id,
+            feature_service_name=feature_service_name,
+            workspace_name=workspace_name,
+        )
+        resp = self._client.post(url=urljoin(self._base_url, "feature-service/metadata"), json=request_data)
```

Comment:
Q: is the `urljoin` worth deriving once per client?

Reply:
Could be; having the endpoints centralized in 1 place also sounds nice.

Reply:
updated.